### PR TITLE
docs(website): added new package creation guide

### PR DESCRIPTION
**Pull request by jakepage91**
## ✅ Checks
- [ ] My code requires changes to the documentation
- [x] I have updated the documentation as required

### CODE REVIEW

**Review by christophenne — COMMENTED** [review]
Overall a nice guide :+1: It covers most parts – the only thing I would add are some additional references to docs here and there (see comments).

**Review by christophenne — COMMENTED** [review]
Overall a nice guide :+1: It covers most parts – the only thing I would add are some additional references to docs here and there (see comments).

File: website/guides/08_package-creation.mdx [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+
+### Step 4: Commit these new changes to your local branch
+
+``` bash
+git add .
+git commit -m "feat(argo-workflow): initial argo-workflow support"
+git push origin add-argo-workflows-support
+```
+
+### Step 5: Test your package integration
```
[the remark this answers is not in the document]
Comment (christophenne) [reply website/guides/08_package-creation.mdx:387]: and vice versa, the contribution guidelines could link to this guide for more detailed instructions

File: website/guides/08_package-creation.mdx [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+```
+
+This is the URL for the public Glasskube package repo which comes configured by default:
+
+``` bash
+NAME       URL                                         DEFAULT  AUTHENTICATION  STATUS  MESSAGE
+glasskube  https://packages.dl.glasskube.dev/packages  Yes      None            Ready   repo has 16 packages
+```
+
+Find the repo [here](https://medium.com/r/?url=https%3A%2F%2Fgithub.com%2Fglasskube%2Fpackages%2Ftree%2Fmain).
```
[the remark this answers is not in the document]
Comment (jakepage91) [reply website/guides/08_package-creation.mdx:54]: Oh yeah to help with markdown formatting I used a tool to generate the MDX from the post on Medium. Totally missed the links that are messed up.

File: website/guides/08_package-creation.mdx [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+![argo-workflow-gif](https://github.com/user-attachments/assets/07123727-9b02-4946-9e98-9feead827f3b)
+
+## Package repository structure 🏗️
+
+![package-repo-structure](https://github.com/user-attachments/assets/67b6d5d5-f8da-4a84-8be9-1ec852ae4507)
+
+**Level 0**
+
+-   **Root packages folder:** where all packages definition files can be found
```
[the remark this answers is not in the document]
Comment (jakepage91) [reply website/guides/08_package-creation.mdx:87]: Does this work? 
**Level 1**

-   **Folder per package**
-   **index.yaml** File which holds important information about each package (updated automatically during CI but manually during testing)

File: website/guides/08_package-creation.mdx [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+
+Expected output:
+
+``` bash
+NAME            URL                                                                                         DEFAULT  AUTHENTICATION  STATUS  MESSAGE
+argo-workflows  https://raw.githubusercontent.com/jakepage91/packages/add-argo-workflows-support/packages/  No       None            Ready   repo has 17 packages
+glasskube       https://packages.dl.glasskube.dev/packages 
+```
+
+### Step 6: Access Glasskube Dashboard and install the package
```
[the remark this answers is not in the document]
Comment (jakepage91) [reply website/guides/08_package-creation.mdx:405]: How about this? 

> Please note that any changes made to your package repository will not be picked up automatically. To ensure all caches are cleared, either wait 5 minutes or restart both the operator and the UI. This is a known issue that will be addressed in a future version.

**Review by pmig — CHANGES_REQUESTED** [review]

File: website/guides/08_package-creation.mdx [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+
+But there is an **elephant** in the room.
+
+> _As a user, you will only adopt a new package manager if it supports the packages you use and if it is easier to use than the alternatives._
+
+As a small team, we are doing our best to support new packages every week. But there are other features and expansions we have planned in the upcoming roadmap that don’t allow us to add new packages to the Glasskube catalogue at a fast enough pace. Luckily, adding new packages to Glasskube is surprisingly straightforward. Through this guide I want to show you just how easily you can add your own packages, both to the Public Glasskube package repo or to your own private repo.
+
+> _In this guide we will learn how to add a Package by integrating the_ [_Argo Workflows_](https://medium.com/r/?url=https%3A%2F%2Fargoproj.github.io%2Fworkflows%2F) _controller._
```
Comment (christophenne) [inline website/guides/08_package-creation.mdx:23]: Please replace the medium link here too.

File: website/guides/08_package-creation.mdx [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+
+An example of hard coding values:
+
+``` yaml
+  values:
+    serviceAccount:
+      name: my-service-account
+```
+
+A current limitation of value definitions is that they can only patch helm values in existing paths, in like the [caddy-ingress-controller](https://medium.com/r/?url=https%3A%2F%2Fgithub.com%2Fcaddyserver%2Fingress%2Fblob%2Fmain%2Fcharts%2Fcaddy-ingress-controller%2Fvalues.yaml%23L33) we instantiate them like so:
```
Comment (christophenne) [inline website/guides/08_package-creation.mdx:193]: Please replace the medium link here too.

File: website/guides/08_package-creation.mdx [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+
+3. Will the package require an **entrypoint** configuration, if so which ports need to be accessible?
+
+4. Will the package require **custom value definitions**, if so which ones?
+
+5. Which **URL references** will be included?
+
+## Let’s integrate Argo Workflows
+
+The best way to learn is by doing so lets integrate a new package. We have already integrated the ArgoCD operator so let’s add the [Argo workflows](https://medium.com/r/?url=https%3A%2F%2Fargoproj.github.io%2Fworkflows%2F) controller too.
```
Comment (christophenne) [inline website/guides/08_package-creation.mdx:297]: Please replace the medium link here too.

**Review by christophenne — CHANGES_REQUESTED** [review]

File: website/sidebar-guides.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ const sidebars: SidebarsConfig = {
           '⚠️ Learn how to install your favorite Kubernetes add-ons using the Glasskube package manager ⚠️',
         slug: '/categories/',
       },
-      items: ['cert-manager', 'ingress-nginx', 'rabbitmq', 'quickwit', 'kube-prom-stack'],
+      items: ['cert-manager', 'ingress-nginx', 'rabbitmq', 'quickwit', 'kube-prom-stack', 'package-creation'],
```
[the remark this answers is not in the document]
Comment (jakepage91) [reply website/sidebar-guides.ts:15]: True, yeah!

**Review by christophenne — APPROVED** [review]

**Review by pmig — APPROVED** [review]

**Pull request merged**